cookieConsent/OneTrust/browser.js

```diff
@@ -0,0 +1,97 @@
+class OneTrust {
+  constructor(sourceConfig) {
+    this.sourceConfig = sourceConfig;
+    // If user does not load onetrust sdk before loading rudderstack sdk
+    // we will not be filtering any of the destinations.
+    if (!window.OneTrust || !window.OnetrustActiveGroups) {
+      logger.debug(
+        `Onetrust window objects not retrieved. Thus events are sent.`
+      );
+      return true;
+    }
+    // OneTrust Cookie Compliance populates a data layer object OnetrustActiveGroups with
+    // the cookie categories that the user has consented to.
+    // Eg: ',C0001,C0003,'
+    // We split it and save it as an array.
+
+    const userSetConsentGroupIds = window.OnetrustActiveGroups.split(","); // Ids user has consented
+
+    // Get information about the cookie script - data includes, consent models, cookies in preference centre, etc.
+    // We get the groups(cookie categorization), user has created in one trust account.
+
+    const oneTrustAllGroupsInfo = window.OneTrust.GetDomainData().Groups;
+    this.userSetConsentGroupNames = [];
+
+    // Get the names of the cookies consented by the user in the browser.
+
+    oneTrustAllGroupsInfo.forEach((group) => {
+      const { CustomGroupId, GroupName } = group;
+      if (userSetConsentGroupIds.includes(CustomGroupId)) {
+        this.userSetConsentGroupNames.push(GroupName.toUpperCase().trim());
+      }
+    });
+  }
+
+  isEnabled(destConfig) {
+    try {
+      /**
+     * Structure of onetrust consent group destination config.
+     * 
+     * "oneTrustConsentGroup": [
+                        {
+                            "oneTrustConsentGroup": "Performance Cookies"
+                        },
+                        {
+                            "oneTrustConsentGroup": "Functional Cookies"
+                        },
+                        {
+                            "oneTrustConsentGroup": ""
+                        }
+                    ]
+      *
+     */
+
+      const { oneTrustCookieCategories } = destConfig; // mapping of the destination with the consent group name
+
+      // If the destination do not have this mapping events will be sent.
+
+      if (!oneTrustCookieCategories) {
+        logger.debug(
+          "No onetrust cookie category set for the destination. Thus events are sent."
+        );
+        return true;
+      }
+
+      // Change the structure of oneTrustConsentGroup as an array and filter values if empty string
+      // Eg:
+      // ["Performance Cookies", "Functional Cookies"]
+
+      const oneTrustConsentGroupArr = oneTrustCookieCategories
+        .map((c) => c.oneTrustCookieCategory)
+        .filter((n) => n);
+      let containsAllConsent = true;
+
+      // Check if any value is there in the destination's cookie groups.
+      // If not events will go anyway to the destination.
+
+      if (
+        Array.isArray(oneTrustConsentGroupArr) &&
+        oneTrustConsentGroupArr.length
+      ) {
```

Comment:
This check is redundant

Reply:
Removing only `Array.isArray(oneTrustConsentGroupArr) &&` this part.

Reply:
Even the length check is not required as the `every` function on an empty array returns true.

Reply:
Okay will remove that

Reply:
done